pkg/servicescm/servicescm.go

```diff
@@ -278,6 +283,48 @@ func (s *Service) hasDependency(possibleDependencies []Service) bool {
 	return false
 }
 
+// hasCycle detects cycles between any of the given services by traversing the resulting dependency graph.
+// Wrapper for Service.hasCycle to handle disconnected graphs
+func hasCycle(services []Service) bool {
```

Comment:
please rename this to something equivalent, would prefer to not have multiple functions with the same name if they are not due to an interface/generics

`validateCycles` perhaps? That way you can also return an error and state which service in particular has the cycle.